tools/scripts/xlf/remove-ee-modules-from-xlf-files.ts

```diff
@@ -0,0 +1,64 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { promisify } from 'util';
+import { TranslationFile, readXLFFile, removeContextsWithPrefix } from './xlf-modifier';
+
+const readdir = promisify(fs.readdir);
+const stat = promisify(fs.stat);
+const xml2js = require('xml2js');
+async function findXlfFiles(directory: string): Promise<string[]> {
+  try {
+    const files = await readdir(directory);
+
+    const xlfFiles = await Promise.all(
+      files.map(async (file) => {
+        const filePath = path.join(directory, file);
+        const fileStats = await stat(filePath);
+
+        if (fileStats.isFile() && file.endsWith('.xlf')) {
+          return filePath;
+        }
+          else {
+          return null;
+        }
+      })
+    );
+
+    // Flatten the array to remove null values and subarray nesting
+    return xlfFiles.filter(Boolean) as string[];
+  } catch (error) {
+    console.error('Error while scanning directory:', error);
+    return [];
+  }
+}
+
+// Example usage
+const directoryPath = 'packages/ui/core/src/locale';
+const modulesToRemoveFromXlf = ['packages/ee/ui-platform','packages/ui/feature-chatbot'];
+findXlfFiles(directoryPath)
+  .then((xlfFiles) => {
+   xlfFiles.forEach(filePath=>{
+        readXLFFile(filePath, function (err:unknown, result:TranslationFile) {
+        if (err) {
+            console.error(err);
+        } else {
+            // Now you can work with the parsed result
+            modulesToRemoveFromXlf.forEach(modulePath=>{
+              const fixedFile =  removeContextsWithPrefix(result, modulePath);
+              var builder = new xml2js.Builder();
+              var xml = builder.buildObject(fixedFile);
+              fs.writeFile(filePath, xml, (err:any) => {
+              if (err) {
+                  console.error('Error writing to file:', err);
+              } else {
+                  console.log('File written successfully');
+              }
+              });
+            });
+
+           
+        }
+        });
+   })
+  })
+  .catch((error) => console.error('Error:', error));

```

Comment:
Currently this code iterates over all xlf files and removes the context/context-groups that contain files from specified modules, I don't know if instead we are just supposed to modifying the main messages.xlf file and then the other locales will be extracted from it instead, if so please let me know.

Reply:
Per Mo, we only generate messages.xlf and then let crowdin generate the other files.